### PR TITLE
Implement modules re-use on config reload

### PR DESCRIPTION
**Pull request by bluca**
When the configuration is reloaded, if a module config has not changed, they can be reused. This was not fully implemented, and these commits finish the work.

### CODE REVIEW

**Comment by bluca** [comment]
Note that, as the individual commits dates suggests, we have been using these changes in production for more than a year now, so they should be very well tested (including QA support) :-)

**Comment by nickbroon** [comment]
@ogasser Ping. Would great to keep this series of split pull requests going.

We have quite a few more PR we hope to raise from all the content in the https://github.com/nickbroon/vermont/commits/brocade branch. This PR gets us about half along the branch. As mentioned before, all of this has been getting regular production QA testing here.

File: src/core/InfoElementCfg.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #ifndef INFOELEMENTCFG_H_
 #define INFOELEMENTCFG_H_
 
+#include <utility>
```
Comment (ogasser) [inline src/core/InfoElementCfg.h]: Do we need this include?

Comment (bluca) [inline src/core/InfoElementCfg.h]: Apparently not, just tried to build without it on Debian 9, can't remember now why it was added
Removed

File: src/modules/ipfix/aggregator/Rule.cpp [hunk 2]
```diff
@@ -516,3 +516,38 @@ int Rule::dataRecordMatches(IpfixDataRecord* record) {
...
+bool operator!=(const Rule &rhs, const Rule &lhs) {
+	return !(lhs == rhs);
+}
+
+bool operator==(const Rule::Field &rhs, const Rule::Field &lhs) {
+	if (rhs.modifier != lhs.modifier) return false;
+	if (rhs.type != lhs.type) return false;
+	if (rhs.pattern == NULL && lhs.pattern == NULL) return true;
+	if (rhs.pattern == NULL || lhs.pattern == NULL) return false;
+	if (rhs.type.length != lhs.type.length) return false;
```
Comment (bluca) [inline src/modules/ipfix/aggregator/Rule.cpp]: That only compares the IeId and IeEnterpriseNumber fields. Should I fold this check in https://github.com/tumi8/vermont/pull/52/files#diff-7018cbd611a4dd66e8daa5523dd3179e ? Or drop it entirely?
I guess if the IDs are the same lengths should be the same too

Comment (bluca) [inline src/modules/ipfix/aggregator/Rule.cpp]: Removed

**Comment by bluca** [comment]
Updated with the requested changes. Thanks for having a look!

**Comment by ogasser** [comment]
Thanks for the changes!

**Comment by bluca** [comment]
Thank you for reviewing and merging! Here's another one: https://github.com/tumi8/vermont/pull/54 :-)

**Pull request merged**